lib/gtl_formatter.rb

```diff
@@ -96,15 +109,33 @@ def self.format_cols(view, row, controller, options)
                 :icon  => icon,
                 :icon2 => icon2,
                 :text  => text}.compact
+      elsif COLUMN_WITH_BACKGROUND_ICON.keys.include?(col)
+        # Generate html for the list icon with background
+        icon = send(COLUMN_WITH_BACKGROUND_ICON[col], record)
+        text = format_col_for_display(view, row, col)
+        item = {:title      => text,
+                :icon       => icon,
+                :background => textual_power_state_whitelisted(record[col]),
+                :text       => text}.compact
       elsif COLUMN_WITH_ICON.keys.include?(col)
         # Generate html for the list icon
         icon = send(COLUMN_WITH_ICON[col], record)
         text = format_col_for_display(view, row, col)
         item = {:title => text,
                 :icon  => icon,
-                :text => text}.compact
+                :text  => text}.compact
       elsif COLUMN_WITH_TIME.include?(col)
         celltext = format_time_for_display(row, col)
+      elsif COLUMN_WITH_OS_ICON.keys.include?(col)
+        @osicon = send(COLUMN_WITH_OS_ICON[col], record)
+        @ostext = format_col_for_display(view, row, col).presence || _(Unknown)
+      elsif COLUMN_WITH_OS_TEXT.include?(col)
```

Comment:
```suggestion
      elsif COLUMN_WITH_OS_TEXT.key?(col)
```

This one is missing `.keys`... was it supposed to be `.keys.include?`. If so, see suggestion above.